scene/resources/material.cpp

```diff
@@ -1273,15 +1273,18 @@ void BaseMaterial3D::_update_shader() {
 	}
 
 	if (distance_fade != DISTANCE_FADE_DISABLED) {
+		// For all distance fade modes, use circular fade (distance to the
+		// object instead of Z distance), so that the fade is always the same
+		// regardless of the camera angle.
+		// This has a small performance cost, but the quality increase is worth it.
```

Comment:
```suggestion
```

This comment won't be helpful after this PR is merged as readers won't be able to see the old code. A comment like this is more appropriate for the commit message and/or the PR itself. As that is what users will see when they traverse the history.

Reply:
Well the comment's verbosity could be reduced a bit but I think it's still useful to clarify that we're using circular fade and not linear on purpose.
Could be:
```
// We use the slight more expensive circular face (distance to the object) instead of linear
// (Z distance), so that the fade is always the same regardless of the camera angle.
```